project-koku/masu
# Remove old temporary files after processing

Closes #141 

**Overview**
Masu downloads cost usage report files to a local temporary directory (i.e. `*.csv.gz`).  Masu's processor will pick up files in this directory to load the file data into the database.  Once the data is loaded masu keeps track of the last completed processing time and from that point on the `.csv.gz` file is no longer needed.  In order to control local storage requirements we need to remove processed CUR files.

This change will remove files under the following criteria:
1. File does not match the current assemblyID
2. File has been processed by masu by confirming that it has a last completed processing timestamp.


**Future work (maybe?)**
For the initial implementation I wanted to keep this simple and also maintain the current file in temporary storage for debugging reasons.  This may turn out to not be necessary and at that point we can easily lift the current assemblyID requirement and it will remove all processed files.  There are also times when we want to remove a file that hasn't been processed but is considered stale since the assemblyID is no longer current.  For the time being I'm not handling this case just to be extra conservative in the deletion policy during the pilot phase.  Also, there is nothing to stop us from logging into the pod and manually deleting temporary data as a backdoor method for freeing up temp space.

**Testing**
1. Use `nise` to load data into S3, download and process with masu.  Verify that the current assemblyID is in temporary storage and the log messages indicates that no files were deleted.
2. Use `nise` to generate another set of data for the same time period as step 1 and verify that the temporary storage now contains only the newest assemblyID and the log message indicates the old file was removed.

Results:
[masu_141_ut.txt](https://github.com/project-koku/masu/files/2306264/masu_141_ut.txt)
